src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -30,14 +30,210 @@
 """
 
 import functools
+import uuid
 import warnings
-from collections.abc import Iterable, Sequence
-from typing import Any
+from collections.abc import Callable, Sequence
+from dataclasses import dataclass
+from typing import Any, Optional
 
-import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
+import pandera as pa
 import scipy.stats
+from dagster import AssetIn, AssetOut, asset, multi_asset
+from pandera.typing import DataFrame, Index, Series
+
+from pudl.logging_helpers import get_logger
+from pudl.metadata.enums import ImputationReasonCodes
+
+logger = get_logger(__file__)
+
+# --- Constants --- #
+
+STANDARD_UTC_OFFSETS: dict[str, str] = {
+    "Pacific/Honolulu": -10,
+    "America/Anchorage": -9,
+    "America/Los_Angeles": -8,
+    "America/Denver": -7,
+    "America/Chicago": -6,
+    "America/New_York": -5,
+    "America/Halifax": -4,
+}
+"""Hour offset from Coordinated Universal Time (UTC) by time zone.
+
+Time zones are canonical names (e.g. 'America/Denver') from tzdata (
+https://www.iana.org/time-zones)
+mapped to their standard-time UTC offset.
+"""
+
+
+# --- Data structures --- #
+
+
+class UTCTimeseriesDataFrame(pa.DataFrameModel):
```

Comment:
These pandera models are primarily for "documentation". They just allow me to add type annotations to various function signatures that give more insight to what is expected as the inputs/outputs than just `pd.DataFrame`